### PR TITLE
Fixing Insert Link Crash

### PR DESCRIPTION
**Pull request by jleandroperez**
### Testing:
1. Launch the Aztec Empty Editor
2. Hit the **Insert Link** button
3. Verify that the app doesn't crash anymore
4. Run the Unit Tests and verify that nothing breaks

Needs Review: @SergioEstevao 
Thanks in advance!!

### CODE REVIEW

File: Aztec/Classes/TextKit/TextView.swift [hunk 1]
```diff
@@ -700,37 +700,48 @@ open class TextView: UITextView {
...
     open func linkURL(forRange range: NSRange) -> URL? {
         let index = maxIndex(range.location)
         var effectiveRange = NSRange()
-        if let attr = storage.attribute(NSLinkAttributeName, at: index, effectiveRange: &effectiveRange) {
-            if let url = attr as? URL {
-                return url
-            } else if let urlString = attr as? String {
-                return URL(string:urlString)
-            }
+        guard index > storage.length,
```
Comment (SergioEstevao) [inline Aztec/Classes/TextKit/TextView.swift]: Maybe here you want ``` index < storage.length```

File: Aztec/Classes/TextKit/TextView.swift [hunk 2]
```diff
@@ -700,37 +700,48 @@ open class TextView: UITextView {
...
     ///
-    /// - Returns: The full Range required by the link.
+    /// - Returns: The full Link's Range.
     ///
     open func linkFullRange(forRange range: NSRange) -> NSRange? {
         let index = maxIndex(range.location)
         var effectiveRange = NSRange()
-        if storage.attribute(NSLinkAttributeName, at: index, effectiveRange: &effectiveRange) != nil {
-            return effectiveRange
+        guard index > storage.length,
```
Comment (SergioEstevao) [inline Aztec/Classes/TextKit/TextView.swift]: Same here: ```index < storage.length```

**Comment by SergioEstevao** [comment]
You may also need to fix the method maxIndex to:
```
func maxIndex(_ index: Int) -> Int {
        if index >= storage.length {
            return max(0,storage.length - 1)
        }

        return index
    }
```

To avoid returning a negative index.

**Comment by jleandroperez** [comment]
@SergioEstevao thanks for the review!! ready for another look sir.

**Comment by jleandroperez** [comment]
Thank you Sergio!!

**Pull request merged**